src/widget/wsearchlineedit.cpp

```diff
@@ -387,17 +388,24 @@ void WSearchLineEdit::keyPressEvent(QKeyEvent* keyEvent) {
         }
         break;
     case Qt::Key_Left:
-    case Qt::Key_Right:
+    case Qt::Key_Right: {
+        // Both keys may change or clear the selection (suggested completion).
+        const bool hadSelectedText = hasSelectedText();
```

Comment:
`hadSelectedTextBeforeKeyPressed` for disambiguation of the very similar identifiers.